Implement "costume name" block

towerofnix opened the pull request:
### Resolves

Resolves #355, since this is the last obsolete block I've spotted. That issue can be reopened if any more are found, though.

Should be merged alongside (incoming blocks PR).

### Proposed Changes

Implements the "costume name" block, which reports the name of the current costume. 

### Reason for Changes

Compatibility with projects that use the block.

### Test Coverage

Added a test for the block, and manually tested with [this project](https://scratch.mit.edu/projects/228377453/). I also changed a variable named "number" to "name" in a related test, since that seems.. more right?

towerofnix committed:
Implement costume name block

towerofnix committed:
Test for getCostumeName

kchadha commented:
@towerofnix Thank you for adding the implementation for this block.
I [commented](https://github.com/LLK/scratch-blocks/pull/1580#issuecomment-396692720) on the discussion in the related scratch-blocks PR, and I think this code can get cleaned up by translating this 2.0 legacy block to the 3.0 `costume [number/name]` block.

towerofnix committed:
Map costumeName -> costumenumbername [name]

...instead of mapping to a new obsolete/hidden block 'costumename'.
Thanks to @kchadha for helping with this!

towerofnix commented:
Thanks, @kchadha, for the info! I didn't spot the `sceneName` -> `backdropnumbername [name]` map. This is a lot cleaner now. I've closed the -blocks PR, because it's not necessary with this.

towerofnix committed:
Remove unused getCostumeName function

kchadha commented:
> I didn't spot the sceneName -> backdropnumbername [name] map.

I didn't either at first! 😄

Thanks for making the change!

kchadha reviewed (approved):
Looks good! Thanks!
